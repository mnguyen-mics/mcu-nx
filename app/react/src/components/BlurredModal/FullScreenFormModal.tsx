import * as React from 'react';
import { createPortal } from 'react-dom';

export interface FullScreenFormModalProps {
  opened?: boolean;
  blurred?: boolean;
}

export default class FullScreenModal extends React.Component<FullScreenFormModalProps> {

  el: HTMLDivElement;

  constructor(props: FullScreenFormModalProps) {
    super(props);
    this.el = document.createElement('div');
  }

  setRootClassName = (remove?: boolean, blurred?: boolean) => {
    const appRoot = document.getElementById('mcs-full-page');
    if (appRoot && !remove && blurred)
      appRoot.classList.add('form-modal-blurred-open')
    if (appRoot && !remove && !blurred)
      appRoot.classList.add('form-modal-open')
    if (appRoot && remove) {
      appRoot.classList.remove('form-modal-blurred-open')
      appRoot.classList.remove('form-modal-open')
    }
  }

  componentDidMount() {
    const {blurred} = this.props;
    const modalRoot = document.getElementById('mcs-edit-modal');
    if (modalRoot)
      modalRoot.appendChild(this.el);
    if (this.props.opened) {
      this.setRootClassName(false, blurred)
    }
  }

  componentDidUpdate(previousProps: FullScreenFormModalProps) {
    const {blurred, opened} = this.props;

    const {opened: previousOpened} = previousProps;

    if (!previousOpened && opened) {
      this.setRootClassName(false, blurred)
    } else if (previousOpened && !opened) {
      this.setRootClassName(true, blurred)
    }
  }
  

  componentWillUnmount() {
    const modalRoot = document.getElementById('mcs-edit-modal');
    if (modalRoot) {
      modalRoot.removeChild(this.el);
    }
    this.setRootClassName(true)
  }


  render() {
    return this.props.opened && createPortal(
      this.props.children,
      this.el,
    );
  }
}
