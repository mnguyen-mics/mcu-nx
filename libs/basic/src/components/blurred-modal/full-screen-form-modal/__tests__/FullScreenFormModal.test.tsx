import 'jest';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import FullScreenFormModal, { FullScreenFormModalProps } from '../FullScreenFormModal';

it('renders the FullScreenModal', () => {
  const props: FullScreenFormModalProps = {
    opened: true,
    blurred: true,
  };

  const component = TestRenderer.create(<FullScreenFormModal {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
