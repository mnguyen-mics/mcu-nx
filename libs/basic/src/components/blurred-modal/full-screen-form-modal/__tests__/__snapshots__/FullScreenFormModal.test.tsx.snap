// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FullScreenModal 1`] = `null`;
